clients/xpert/xrtcomponent.go

```diff
@@ -27,3 +28,20 @@ func (c *xrtClient) UpdateLuaScript(ctx context.Context, luaScript string) error
 	}
 	return nil
 }
+
+func (c *xrtClient) DiscoverComponents(ctx context.Context, category string, subscribeTimeout time.Duration) ([]xrtmodels.MultiComponentsResponse, errors.EdgeX) {
+	request := xrtmodels.NewComponentDiscoverRequest(clientName, category)
+	var response []xrtmodels.MultiComponentsResponse
+
+	err := c.sendXrtRequestWithSubTimeout(ctx, c.requestTopic, request.RequestId, request, &response, subscribeTimeout)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.Kind(err), "failed to discover the xrt components", err)
+	}
+
+	for _, compResp := range response {
+		if compResp.Result.Error() != nil {
+			return nil, compResp.Result.Error()
```

Comment:
ignore the individual component result error, so the for loop should be removed
the caller should handle those errors

Reply:
Done